sherpa/cpp_api/offline_recognizer.cc

```diff
@@ -116,17 +116,20 @@ OfflineRecognizer::~OfflineRecognizer() = default;
 
 std::vector<OfflineRecognitionResult> OfflineRecognizer::DecodeFileBatch(
     const std::vector<std::string> &filenames) {
+  torch::NoGradGuard no_grad;
```

Comment:
Caution: `NoGradGuard` is thread-local. 

It is not enough to add it only to the main thread.